test/images/agnhost/liveness/server.go

```diff
@@ -54,6 +60,13 @@ func main(cmd *cobra.Command, args []string) {
 			w.WriteHeader(200)
 			w.Write([]byte("ok"))
 		}
+		if disconnect {
```

Comment:
do you need this disconnect for both - successful and failed probes? If so - maybe you should lead with this option and return after

Reply:
Yes. I need to disconnect after a header is sent for both cases, which are mentioned in #122591.